packages/two_dimensional_scrollables/lib/src/table_view/table.dart

```diff
@@ -0,0 +1,1029 @@
+    required super.horizontalAxisDirection,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required super.delegate,
+    required super.mainAxis,
+    required super.childManager,
+    super.cacheExtent,
+    super.clipBehavior,
+  });
+
+  TableCellDelegateMixin get _delegate => delegate as TableCellDelegateMixin;
+
+  // Cached Table metrics
+  Map<int, _Span> _columnMetrics = <int, _Span>{};
+  Map<int, _Span> _rowMetrics = <int, _Span>{};
+  int? _firstRegularRow;
+  int? _firstRegularColumn;
+  int? _lastRegularRow;
+  int? _lastRegularColumn;
+
+  TableVicinity? get _firstRegularCell {
+    if (_firstRegularRow == null || _firstRegularColumn == null) {
+      return null;
+    }
+    return TableVicinity(column: _firstRegularColumn!, row: _firstRegularRow!);
+  }
+
+  TableVicinity? get _lastRegularCell {
+    if (_lastRegularRow == null || _lastRegularColumn == null) {
+      return null;
+    }
+    return TableVicinity(column: _lastRegularColumn!, row: _lastRegularRow!);
+  }
+
+  int? get _lastPinnedRow =>
+      _delegate.pinnedRowCount > 0 ? _delegate.pinnedRowCount - 1 : null;
+  int? get _lastPinnedColumn =>
+      _delegate.pinnedColumnCount > 0 ? _delegate.pinnedColumnCount - 1 : null;
+
+  double get _pinnedRowsExtent => _lastPinnedRow != null
+      ? _rowMetrics[_lastPinnedRow]!.trailingOffset
+      : 0.0;
+  double get _pinnedColumnsExtent => _lastPinnedColumn != null
+      ? _columnMetrics[_lastPinnedColumn]!.trailingOffset
+      : 0.0;
+
+  @override
+  TableViewParentData parentDataOf(RenderBox child) =>
+      child.parentData! as TableViewParentData;
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! TableViewParentData) {
+      child.parentData = TableViewParentData();
+    }
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    RenderBox? cell = firstChild;
+    while (cell != null) {
+      final TableViewParentData cellParentData = parentDataOf(cell);
+      if (!cellParentData.isVisible) {
+        // This cell is not visible, so it cannot be hit.
+        cell = childAfter(cell);
+        continue;
+      }
+      final Rect cellRect = cellParentData.paintOffset! & cell.size;
+      if (cellRect.contains(position)) {
+        result.addWithPaintOffset(
+          offset: cellParentData.paintOffset,
+          position: position,
+          hitTest: (BoxHitTestResult result, Offset transformed) {
+            assert(transformed == position - cellParentData.paintOffset!);
+            return cell!.hitTest(result, position: transformed);
+          },
+        );
+        switch (mainAxis) {
+          case Axis.vertical:
+            // Row major order, rows go first.
+            result.add(
+                HitTestEntry(_rowMetrics[cellParentData.tableVicinity.row]!));
+            result.add(HitTestEntry(
+                _columnMetrics[cellParentData.tableVicinity.column]!));
```

Comment:
trailing comma should format these nicer (here and elsewhere)

```suggestion
            result.add(
                HitTestEntry(_rowMetrics[cellParentData.tableVicinity.row]!),);
            result.add(HitTestEntry(
                _columnMetrics[cellParentData.tableVicinity.column]!),);
```